staging/src/k8s.io/kubelet/pkg/apis/podresources/v1/api.proto

```diff
@@ -77,3 +79,35 @@ message TopologyInfo {
 message NUMANode {
     int64 ID = 1;
 }
+
+// DynamicResource contains information about the devices assigned to a container by DRA
+message DynamicResource {
+    string class_name = 1;
+    string claim_name = 2;
+    string claim_namespace = 3;
+    repeated PluginCDIDevices plugins_cdi_devices = 4;
```

Comment:
another option is to change this to map<string, PluginCDIDevices> plugins_cdi_devices and the string can be the kubelet plugin name.

Reply:
The issue is that I know of upcoming uses where resource claims will be used with different types of plugins that do not return CDIDevices at all (bother rather a different type of resource). I want to future proof for them as well in a wrapping struct that (in the future) could have a type field and may multiple optional fields depending on the type (with CDIDevices being one of them).

Reply:
What about:
```
message DynamicResource {
    string class_name = 1;
    string claim_name = 2;
    string claim_namespace = 3;
    repeated ClaimResource claim_resources = 4;
}

message ClaimResource {
    repeated CDIDevice cdi_devices = 1;
    // add other possible fields here in the future
}
```

Reply:
So something like this:
message DynamicResource {
    string class_name = 1;
    string claim_name = 2;
    string claim_namespace = 3;
    repeated Resources resources = 4;
}

message Resources {
  oneof resource{
        CDIDevices cdi_devices = 1;
  }    
}

message CDIDevices {
  repeated CDIDevice cdi_devices = 1;
}

Reply:
`oneof` is not allowed in core k8s code

Reply:
You can find it in vendored code, but not within core kubernetes itself.

Reply:
Actually, I can't find any reference that says it's explicitly disallowed, I just don't know of any code that uses it (and a simple grep turns up nothing).

Reply:
by the way the ClaimResource  is per plugin as I doing https://github.com/kubernetes/kubernetes/pull/115847/files/d5fc1e401a80ccc69f2a7318d08621f57c48cddf#diff-c07d343a8d459721656cf7ddacda71506bde01a5cac288d15a4aba6bfc9a1caeR987 ?